### PR TITLE
Deprecate Map, List, and Set in org.opensearch.common.collect (#6609)

### PR DESCRIPTION
**Pull request by diliproshitha**
### Description
Deprecate following classes and replace usages with java.util classes
- org.opensearch.common.collect.Map
- org.opensearch.common.collect.List
- org.opensearch.common.collect.Set

### Issues Resolved
[#6609 ]

### Check List
- [ X ] All tests pass
- [ X ] Commits are signed per the DCO using --signoff
- [ X ] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by kotwanikunal** [comment]
@diliproshitha Can you please execute `./gradlew spotlessApply` and `./gradlew precommit`? More on this [here](https://github.com/opensearch-project/OpenSearch/blob/main/DEVELOPER_GUIDE.md)

File: libs/common/src/main/java/org/opensearch/common/collect/List.java [hunk 1]
```diff
@@ -37,10 +37,10 @@
 /**
  * Java 9 List
  *
- * todo: deprecate and remove w/ min jdk upgrade to 11?
  *
  * @opensearch.internal
  */
+@Deprecated
```
Comment (reta) [inline libs/common/src/main/java/org/opensearch/common/collect/List.java:43]: We could go even futher: `@Deprecated(forRemoval = true)`

**Comment by diliproshitha** [comment]
> @diliproshitha Can you please execute `./gradlew spotlessApply` and `./gradlew precommit`? More on this [here](https://github.com/opensearch-project/OpenSearch/blob/main/DEVELOPER_GUIDE.md)

This is done, and changes have been committed.

File: libs/common/src/main/java/org/opensearch/common/collect/List.java (hunk 1 above)
Comment (reta) [reply libs/common/src/main/java/org/opensearch/common/collect/List.java]: :+1:

**Review by reta — APPROVED** [review]

**Comment by kotwanikunal** [comment]
```
* What went wrong:
Execution failed for task ':libs:opensearch-common:compileTestJava'.
> Compilation failed; see the compiler error output for details.
```

**Comment by diliproshitha** [comment]
> ```
> * What went wrong:
> Execution failed for task ':libs:opensearch-common:compileTestJava'.
> > Compilation failed; see the compiler error output for details.
> ```

This is fixed.

**Review by kotwanikunal — APPROVED** [review]

**Comment by kotwanikunal** [comment]
Thanks for your contribution @diliproshitha!

**Comment by kotwanikunal** [comment]
@diliproshitha Adding in these changes to main: #6687 @reta

**Pull request merged**